src/psyclone/gocean1p0.py

```diff
@@ -880,6 +881,222 @@ def backward_dependence(self):
         return None
 
 
+class GOStencil(object):
+    '''GOcean 1.0 stencil information for a kernel argument as obtained by
+    parsing the kernel meta-data. The expected structure of the
+    metadata and its meaning is provided in the description of the
+    load method
+
+    '''
+    def __init__(self):
+        ''' Set up any internal variables. '''
+        self._has_stencil = None
+        self._stencil = [[0 for _ in range(3)] for _ in range(3)]
+        self._name = None
+        self._initialised = False
+
+    def load(self, stencil_info, kernel_name):
+        '''Take parsed stencil metadata information, check it is valid and
+        store it in a convenient form. The kernel_name argument is
+        only used to provide the location if there is an error.
+
+        The stencil information should either be a name which
+        indicates a particular type of stencil or in the form
+        stencil(xxx,yyy,zzz) which explicitly specifies a stencil
+        shape where xxx, yyy and zzz are triplets of integers
+        indicating whether there is a stencil access in a particular
+        direction and the depth of that access. For example:
+
+        stencil(010,  !   N
+                212,  !  W E
+                010)  !   S
+
+        indicates that there is a stencil access of depth 1 in the
+        "North" and "South" directions and and stencil access of depth
+        1 and 2 in the "East" and "West" directions. The value at the
+        centre of the stencil will not be used by PSyclone but can be
+        0 or 1 and indicates whether the local field value is
+        accessed.
+
+        The convention is for the associated arrays to be
+        2-dimensional. If we denote the first dimension as "i" and the
+        second dimension as "j" then the following directions are
+        assumed:
+
+        j
+        ^
+        |
+        |
+        ---->i
+
+        For example a stencil access like:
+
+        a(i,j) + a(i+1,j) + a(i,j-1)
+
+        would be stored as:
+
+        stencil(000,
+                011,
+                010)
+
+        :param stencil_info: contains the appropriate part of the parser AST
+        :type stencil_info: :py:class:`psyclone.expression.FunctionVar`
+        :param string kernel_name: the name of the kernel from where
+        this stencil information came from.
+        :raises ParseError: if the supplied stencil information is invalid
+
+        '''
+        self._initialised = True
+        if isinstance(stencil_info, expr.FunctionVar):
```

Comment:
I'd prefer it if you raised the exception here `if not isinstance(stencil_info, expr.FunctionVar)`. You can then immediately see (in the code) what happens if it's not right.

Reply:
I agree. It also reduces the indentation